lib/tasks/data_hygiene.rake

```diff
@@ -65,15 +66,44 @@ namespace :data_hygiene do
           new_attr = DataHygiene::AttachmentAttributeUpdater.call(attachment, dry_run: dry_run)
           if old_attr == new_attr
             puts "Attachment ID #{attachment.id} command paper number is already valid. Skipping..."
-          elsif dry_run
-            puts "Attachment ID #{attachment.id} command paper number would have changed from #{old_attr} to #{new_attr}"
           else
-            puts "Updated attachment ID #{attachment.id} command paper number from #{old_attr} to #{new_attr}"
+            affected_attachments << attachment
+            if dry_run
+              puts "Attachment ID #{attachment.id} command paper number would have changed from #{old_attr} to #{new_attr}"
+            else
+              puts "Updated attachment ID #{attachment.id} command paper number from #{old_attr} to #{new_attr}"
+            end
           end
         rescue DataHygiene::AttachmentAttributeNotFixable
           puts "Attachment ID #{attachment.id} command paper number cannot be fixed automatically: #{old_attr}"
         end
       end
+
+      puts "Attachments updated: #{affected_attachments.map(&:id).join(', ')}"
+
+      edition_ids = affected_attachments.map do |attachment|
+        case attachment.attachable_type
+        when "Edition"
+          attachment.attachable_id
+        when "Response"
+          Response.where(id: attachment.attachable_id).pluck(:edition_id).first
+        else
+          raise "Unknown attachable type #{attachment.attachable_type}"
```

Comment:
This isn't ideal, as if this happens during the real run, we could lose all the data about attachments already changed, unless we'd previously done a dry run and saved it. As long as we're doing that, I guess it doesn't matter. We could always put this whole task in a transaction, as a quick fix for that.

Reply:
I see L82 also accounts for that somewhat, as well 👍

Reply:
That's fair 👍 L82 wouldn't be reached if the exception is raised, and we wouldn't know which attachments we've updated if we hadn't run a dry run first. I'll look at putting all this in a transaction.